src/net_processing.cpp

```diff
@@ -534,6 +549,8 @@ void PeerLogicValidation::FinalizeNode(NodeId nodeid, bool& fUpdateConnectionTim
     nPreferredDownload -= state->fPreferredDownload;
     nPeersWithValidatedDownloads -= (state->nBlocksInFlightValidHeaders != 0);
     assert(nPeersWithValidatedDownloads >= 0);
+    g_outbound_peers_with_protect_from_disconnect -= state->m_protect_from_disconnect;
```

Comment:
`int -= bool`?

Reply:
More verbose but IMO more clear:
```cpp
if (state->m_protect_from_disconnect) {
    assert(g_outbound_peers_with_protect_from_disconnect > 0);
    --g_outbound_peers_with_protect_from_disconnect;
}
```

Reply:
I was following the pattern already in use in the lines above (`nPreferredDownload -= state->fPreferredDownload`).  But I can change to something like what you suggest if this is frowned upon for some reason.